src/core.jl

```diff
@@ -314,8 +311,30 @@ julia> common_neighbors(g, 1, 4)
  3
 ```
 """
-common_neighbors(g::AbstractGraph, u::Integer, v::Integer) =
-    intersect(neighbors(g, u), neighbors(g, v))
+function common_neighbors(g::AbstractGraph, u::Integer, v::Integer)
+    common = Vector{Int}()
```

Comment:
Should be `Vector{eltype(g)}`. Can you please also add a test, that verifies that the result has the same eltype as the input graph?